src/parsers/perf/perfparser.cpp

```diff
@@ -1369,6 +1367,7 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
             }
         });
 
+        d.disassemblyResult = m_disassemblyResult;
```

Comment:
why this move?

Reply:
I can not remove  disassemblyResult.arch = QString::fromUtf8(features.arch); . At least when architecture was not set through UI it should be computed automatically.

Reply:
I don't get that at all - the parser should know about the arch that was referenced in the `perf.data` file itself and then just pass that through. Then in the UI we can use that data to map it to a `objdump` binary, why would we need to pass an arch *into* the parser?!